src/OrderFormAcceptanceTests.Steps/Steps/AssociatedServices.cs

```diff
@@ -152,25 +153,55 @@ public void GivenFillsInTheAssociatedServiceEditFormWithValidData()
         }
 
         [Given(@"an Associated Service with a flat price variable \(On-demand\) order type with the quantity period per year is saved to the order")]
-        public void GivenAnAssociatedServiceWithAFlatPriceVariableOn_DemandOrderTypeWithTheQuantityPeriodPerYearIsSavedToTheOrder()
+        public async Task GivenAnAssociatedServiceWithAFlatPriceVariableOn_DemandOrderTypeWithTheQuantityPeriodPerYearIsSavedToTheOrder()
         {
             SetOrderAssociatedServicesSectionToComplete();
-            var orderItem = OrderItem.GenerateAssociatedServiceWithFlatPricedVariableOnDemand((Order)Context[ContextKeys.CreatedOrder]);
-            orderItem.EstimationPeriodId = TimeUnit.Year;
-            orderItem.Create(Test.OrdapiConnectionString);
-            Context.Add(ContextKeys.CreatedOrderItem, orderItem);
+            var order = (Order)Context[ContextKeys.CreatedOrder];
+            var orderItem = await OrderItemHelper.CreateOrderItem(
+                order,
+                CatalogueItemType.AssociatedService,
+                CataloguePriceType.Flat,
+                ProvisioningType.OnDemand,
+                DbContext,
+                Test.BapiConnectionString);
+
+            var recipients = await ServiceRecipientHelper.Generate(order.OrderingParty.OdsCode, Test.OdsUrl);
+
+            await OrderItemHelper.AddRecipientToOrderItem(orderItem, recipients, DbContext);
+
+            order.AddOrUpdateOrderItem(orderItem);
+
+            DbContext.Update(order);
```

Comment:
I suspect you may not actually need to call `Update` here but that's something you can review later (same applies elsewhere).